app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -225,6 +226,15 @@ private void requestFavor() {
         o -> {
           setupFavorListener(getView(), currentFavor.getId());
           CommonTools.showSnackbar(currentView, getString(R.string.favor_request_success_msg));
+
+          // update user info
+          UserUtil.getSingleInstance()
+              .findUser(DependencyFactory.getCurrentFirebaseUser().getUid())
+              .thenAccept(
+                  user -> {
+                    user.setRequestedFavors(user.getRequestedFavors() + 1);
+                    UserUtil.getSingleInstance().updateUser(user);
```

Comment:
Should maybe user a UserViewModel if the FavorViewModel is getting too large. I think we should try to use MVVM just to make testing easier (single dependency)

Reply:
Yes I think using a ViewModel for the User would make sense. Maybe we should also discuss about a way to structure better the dependencies but I agree we can leave it for the next sprints